src/Schema/PostgreSQLSchemaManager.php

```diff
@@ -135,7 +135,13 @@ protected function _getPortableTableForeignKeyDefinition(array $tableForeignKey)
      */
     protected function _getPortableViewDefinition(array $view): View
     {
-        return new View($view['schemaname'] . '.' . $view['viewname'], $view['definition']);
+        if ($view['schemaname'] === $this->getCurrentSchema()) {
```

Comment:
The same rule is already used for table and sequence introspection: https://github.com/doctrine/dbal/blob/aa7f89343057f63545b27dd0ddf3f4b54c343fee/src/Schema/PostgreSQLSchemaManager.php#L144-L149

Technically, this is a bug fix (because the existing behavior is inconsistent with the rest of the library) but fixing it would be a BC break, so we cannot backport it. Also, the support for view in DBAL is rudimentary, so I don't think it's a real problem in 4.x.